php/php_storybook/scripts/ecl-diff.js

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env node
+
+/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, no-console,
+no-param-reassign, no-restricted-syntax, no-await-in-loop */
+
+const fs = require('fs');
+const htmlDiffer = require('html-differ');
+const logger = require('html-differ/lib/logger');
+const puppeteer = require('puppeteer');
+const decode = require('decode-html');
+const yargsInteractive = require('yargs-interactive');
+let packages = require('../../../../src/ec/.storybook/ec-packages.js').list;
+
+const system = 'ec';
+const rootFolder = process.cwd();
+const distFolder = `${rootFolder}/php`;
+const systemFolder = `${distFolder}/packages/${system}`;
+const getBase = element => {
+  [, element] = element.split('ec-component-');
+  return element;
+};
+// We build a list of components by their root name.
+packages = packages.map(getBase);
+packages.pop();
+
+// The questions we need an answer for.
+const options = {
+  interactive: { default: true },
+  component: {
+    type: 'list',
+    choices: packages,
+    describe: 'The component you want to diff',
+  },
+  variant: {
+    type: 'input',
+    default: '',
+    describe: 'The name of the twig variant (what is after the -- Ex: audio) ',
+  },
+  language: {
+    type: 'list',
+    describe: 'Choose whether to use the file rendered via js or php',
+    default: 'php',
+    prompt: 'always',
+    choices: ['php', 'js'],
+  },
+  eclSection: {
+    type: 'list',
+    describe: 'The group the component is associated with in ECL',
+    choices: ['components', 'page-structure'],
+  },
+  eclSubSection: {
+    type: 'list',
+    describe: 'Is the component nested into a sub-section?',
+    default: 'none',
+    prompt: 'always',
+    choices: ['none', 'forms', 'navigation', 'banners'],
```

Comment:
I think you should add 'footers', 'site headers', 'page headers', 'banners'?

Reply:
Why @papegaill? banners is already in place, site-headers and page-headers is not anything in the "path" of the ecl story, the site-header-core for instance you reach it at: components-site-headers-core--default, so there is no sub-section for them. ( the same for page headers) ;)